uast/diff/diff.go

```diff
@@ -0,0 +1,287 @@
+package diff
+
+import (
+	"fmt"
+	"github.com/heetch/lapjv"
+	"gopkg.in/bblfsh/sdk.v2/uast/nodes"
+)
+
+type decisionType interface {
+	isDecisionType()
+	cost() int
+}
+
+type basicDecision struct {
```

Comment:
This can be also inlined to every decision type. Yes, it will require to write `cost()` function for each type, but it will be easier to work with decisions: `replaceDecision{cost}` instead of `replaceDecision{basicDecision{cost}}`.